api/v1alpha1/addon_types.go

```diff
@@ -94,69 +97,79 @@ type NFD struct {
 type ClusterAutoscaler struct {
 	// Addon strategy used to deploy cluster-autoscaler to the management cluster
 	// targeting the workload cluster.
+	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:Enum=ClusterResourceSet;HelmAddon
 	Strategy AddonStrategy `json:"strategy"`
 }
 
 type DefaultStorage struct {
 	// Name of the CSI Provider for the default storage class.
+	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:Enum=aws-ebs;nutanix
 	ProviderName string `json:"providerName"`
+
 	// Name of storage class config in any of the provider objects.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
 	StorageClassConfigName string `json:"storageClassConfigName"`
 }
 
 type CSI struct {
-	// +optional
+	// +kubebuilder:validation:Optional
 	Providers []CSIProvider `json:"providers,omitempty"`
-	// +optional
+
+	// +kubebuilder:validation:Optional
 	DefaultStorage *DefaultStorage `json:"defaultStorage,omitempty"`
 }
 
 type CSIProvider struct {
 	// Name of the CSI Provider.
+	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:Enum=aws-ebs;nutanix
 	Name string `json:"name"`
 
-	// +optional
+	// +kubebuilder:validation:Optional
 	StorageClassConfig []StorageClassConfig `json:"storageClassConfig,omitempty"`
 
 	// Addon strategy used to deploy the CSI provider to the workload cluster.
+	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:Enum=ClusterResourceSet;HelmAddon
 	Strategy AddonStrategy `json:"strategy"`
 
 	// The reference to any secret used by the CSI Provider.
-	// +optional
-	Credentials *corev1.LocalObjectReference `json:"credentials,omitempty"`
+	// +kubebuilder:validation:Optional
+	Credentials *LocalObjectReference `json:"credentials,omitempty"`
 }
 
 type StorageClassConfig struct {
 	// Name of storage class config.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
 	Name string `json:"name"`
 
 	// Parameters passed into the storage class object.
-	// +optional
+	// +kubebuilder:validation:Optional
 	Parameters map[string]string `json:"parameters,omitempty"`
 
+	// +kubebuilder:validation:Optional
 	// +kubebuilder:validation:Enum=Delete;Retain;Recycle
 	// +kubebuilder:default=Delete
-	// +optional
 	ReclaimPolicy corev1.PersistentVolumeReclaimPolicy `json:"reclaimPolicy,omitempty"`
 
 	// +kubebuilder:validation:Enum=Immediate;WaitForFirstConsumer
+	// +kubebuilder:validation:Optional
 	// +kubebuilder:default=WaitForFirstConsumer
-	// +optional
 	VolumeBindingMode storagev1.VolumeBindingMode `json:"volumeBindingMode,omitempty"`
```

Comment:
same as above. should be pointer?

Reply:
This is a slice (aliased type) so is effectively already a pointer.